### PR TITLE
Move map logic into seperate map class

### PR DESCRIPTION
**Pull request by dolanmiu**

### CODE REVIEW

File: src/main/client/scene/game-scene.ts [hunk 1]
```diff
@@ -8,24 +8,16 @@ import { Bullets } from './default-bullet';
...
 import Vector2 = Phaser.Math.Vector2;
-
-interface Color {
-   readonly red: number;
-   readonly green: number;
-   readonly blue: number;
-   readonly alpha: number;
-}
-
-const MOON_RADIUS = 200;
```
Comment (dolanmiu) [inline src/main/client/scene/game-scene.ts]: Now it uses the canvas.width rather than this value :)

File: src/main/client/scene/game-scene.ts [hunk 2]
```diff
@@ -177,7 +130,11 @@ export class GameScene extends Scene {
       if (this.cursorKeys.left.isDown) {
          for (let _ = 0; _ < this.maxHorizontalSpeed; _++) {
             if (
-               !this.hitTestTerrain(this.character.x - 1, this.character.y, this.createLocalWall(this.character, 10))
+               !this.mapSprite.hitTestTerrain(
```
Comment (dolanmiu) [inline src/main/client/scene/game-scene.ts:133]: very strange, my prettier does this

File: src/main/client/scene/game-scene.ts [hunk 3]
```diff
@@ -41,14 +33,14 @@ export class GameScene extends Scene {
...
       this.mapComponent.mapLoaded$.subscribe((canvas) => {
-         this.terrainTexture = this.textures.addCanvas('terrain', canvas);
-         this.add.sprite(0, 0, 'terrain');
+         this.mapSprite = new MapSprite({
+            scene: this,
+            canvas: canvas,
+         });
       });
-      this.mapComponent.updated$.subscribe(() => this.terrainTexture.update());
+      this.mapComponent.updated$.subscribe(() => this.mapSprite.update());
```
Comment (dolanmiu) [inline src/main/client/scene/game-scene.ts]: We might have to call update() on the underlying texture, but we will see

File: src/main/client/map/map-sprite.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import * as Phaser from 'phaser';
```
Comment (dolanmiu) [inline src/main/client/map/map-sprite.ts:1]: All collision logic is in here 👍

Comment (barnabasbartha) [reply src/main/client/map/map-sprite.ts:1]: Nice one!

**Review by barnabasbartha — APPROVED** [review]

**Pull request merged**